api/lib/opentelemetry/trace/trace_flags.rb

```diff
@@ -16,10 +16,9 @@ class << self
         # Returns a newly created {TraceFlags} with the specified flags.
         #
         # @param [Integer] flags 8-bit byte of bit flags
-        # @raise [ArgumentError] If flags is not an 8-bit byte
         # @return [TraceFlags]
         def from_byte(flags)
-          raise ArgumentError, 'flags must be an 8-bit byte' unless flags & ~0xFF == 0 # rubocop:disable Style/NumericPredicate
+          flags = 0 unless flags & ~0xFF == 0 # rubocop:disable Style/NumericPredicate
```

Comment:
This will blow up with `NoMethodError` if passed, e.g., a String. Are we OK with that? Technically, while this is a public API, I expect it to only be called by instrumentation (specifically things extracting trace context in various formats from the wire), most of which will live in the SDK & API.